File: 프로그래머스/unrated/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
# def solution(data, ext, val_ext, sort_by):
#     answer = []
#     for i in data:
#         if ext == "code":
#             if i[0] < val_ext:
#                 answer.append(i)
#         elif ext == "date":
#             if i[1] < val_ext:
#                 answer.append(i)
#         elif ext == "maximum":
#             if i[2] < val_ext:
#                 answer.append(i)
#         else:
#             if i[3] < val_ext:
#                 answer.append(i)
                
#         if sort_by == "code":
#             answer.sort(key=lambda x:x[0])
#         elif sort_by == "date":
#             answer.sort(key=lambda x:x[1])
#         elif sort_by == "maximum":
#             answer.sort(key=lambda x:x[2])
#         else:
#             answer.sort(key=lambda x:x[3])
                
#     return answer

def solution(data, ext, val_ext, sort_by):
    answer = []
    
    idx = ["code", "date", "maximum", "remain"]
    ext_idx = idx.index(ext)
    sort_idx = idx.index(sort_by)

    for i in data:
        if i[ext_idx] < val_ext:
            answer.append(i)

    answer.sort(key = lambda x: x[sort_idx])
    
    return answer

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;

class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        ArrayList<int[]> lst = new ArrayList<>();
        int idx = -1;
        if (ext.equals("code")) idx = 0;
        else if (ext.equals("date")) idx = 1;
        else if (ext.equals("maximum")) idx = 2;
        else idx = 3;
        
        for (int i = 0; i < data.length; i++) {
            if (data[i][idx] < val_ext) lst.add(data[i]);
        }
        
        int sort_idx = -1;
        if (sort_by.equals("code")) sort_idx = 0;
        else if (sort_by.equals("date")) sort_idx = 1;
        else if (sort_by.equals("maximum")) sort_idx = 2;
        else sort_idx = 3;
        
        final int sortIdx = sort_idx;
        
        lst.sort(new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return a[sortIdx] - b[sortIdx];
            }
        });
        
        return lst.toArray(new int[0][]);
    }
}